### PR TITLE
fix: fix delay clean logic when multiple healthchecker was started

### PR DESCRIPTION
**Pull request by oowl**
backport https://github.com/Kong/lua-resty-healthcheck/pull/146

FTI-5478

### CODE REVIEW

**Review by locao — DISMISSED** [review]

**Review by outsinre — APPROVED** [review]

File: .github/workflows/old_os.yml [hunk 1]
```diff
@@ -1,189 +0,0 @@
-name: Build and test for Ubuntu 18.04
```
Comment (windmgc) [inline .github/workflows/old_os.yml:1]: Could you explain why this file is being removed?

**Review by windmgc — APPROVED** [review]
Approved with one minor question

File: .github/workflows/old_os.yml (hunk 1 above)
Comment (oowl) [reply .github/workflows/old_os.yml:1]: Because GitHub action no longer supports ubuntu 18.04. This action can not be executed now.

Comment (windmgc) [reply .github/workflows/old_os.yml:1]: Ah okay

Comment (ms2008) [reply .github/workflows/old_os.yml:1]: There seems to be a workaround for this problem. Ref: https://github.com/LuaLS/lua-language-server/pull/1835/files

**Review by ms2008 — APPROVED** [review]

**Pull request merged**